runtime/userspace/api/spdm-lib/src/commands/finish_rsp.rs

```diff
@@ -53,13 +50,42 @@ impl FinishRspBase {
         Self {
             _reserved0: 0,
             _reserved1: 0,
-            opaque_data_len: 0,
         }
     }
 }
 
+async fn verify_requester_verify_data(
+    ctx: &mut SpdmContext<'_>,
+    session_id: u32,
+    requester_verify_data: &[u8; SHA384_HASH_SIZE],
+    req_payload: &mut MessageBuf<'_>,
+) -> CommandResult<()> {
+    // Compute TH1 transcript hash for generating the session handshake key
+    let th1_transcript_hash = ctx
+        .transcript_hash(TranscriptContext::Th, Some(session_id), false)
+        .await?;
+
+    // generate session handshake key
+    let session_info = ctx
+        .session_mgr
+        .session_info_mut(session_id)
+        .map_err(|e| (false, CommandError::Session(e)))?;
+
+    let computed_hmac = session_info
+        .compute_hmac(SessionKeyType::RequestFinishedKey, &th1_transcript_hash)
+        .await
+        .map_err(|e| (false, CommandError::Session(e)))?;
+
+    if computed_hmac != *requester_verify_data {
```

Comment:
For hashes and HMACs we should use a constant-time comparison function, like the `constant_time_eq` crate.